File: nodejs/expressmvc/routes/open-pades-express.js
const express = require('express');
const {
    PadesSignatureExplorer,
    StandardSignaturePolicies
} = require('pki-express');
const { StorageMock } = require('../storage-mock');
const { Util } = require('../util');

const router = express.Router();

/**
 * GET /open-pades
 *
 * This route only submits a PDF file to PKI Express for inspection of its signatures.
 */

router.get('/', (req, res, next) => {
	// Get parameters from url
    const fileId = req.query['fileId'];
    
	// Verify if the provided fileId exists.
	if (!StorageMock.existsSync({fileId: fileId})) {
		let notFound = new Error('The fileId was not found');
		notFound.status = 404;
		next(notFound);
		return;
    }
    filePath = StorageMock.getDataPath(fileId);

    // Get an instance of PadesSignatureExplorer class, used to open/validate PDF
    // signatures.
    sigExplorer = new PadesSignatureExplorer();
    Util.setPkiDefaults(sigExplorer);

    // Specify that we want to validate the signatures in the file, not only
    // inspect them.
    sigExplorer.validate = true;

    // Set the PDF file.
    sigExplorer.setSignatureFileFromPathSync(filePath);

    // Specify the parameters for the signature validation:
    // Accept any PAdES signature as long as the signer has an ICP-Brasil
    // certificate.
    sigExplorer.signaturePolicy = StandardSignaturePolicies.PADES_BASIC;

    // Specify the security context to be used to determine trust in the
    // certificate chain. We have encapsulated the security context on util.js.
    sigExplorer.securityContextId = Util.getSecurityContextId(res.locals.environment);

    // Call the open() method, which returns the signature file's information.
    sigExplorer.open().then(r=> res.render('open-pades-express', { signature: r }));
});

module.exports = router;

File: python/flask/sample/views/open_pades_express.py
import os

from flask import Blueprint
from flask import render_template
from flask import current_app
from pkiexpress import PadesSignatureExplorer

from sample.utils import set_pki_defaults

# 26-08-2022
# By further inspecting in the latest Blueprint documentation (https://flask.palletsprojects.com/en/2.2.x/api/#blueprint-objects), 
# when creating a Blueprint object, the first parameter (name) is prepend to the URL endpoint. Therefore, Blueprint no longer 
# allows dots in the name since it would break the URL entirely.
__name__ = __name__.replace(".","/")
blueprint = Blueprint(os.path.basename(__name__), __name__,
                      url_prefix='/open-pades-express')


@blueprint.route('/<file_id>')
def index(file_id):

    # Verify if the provided userfile exists.
    file_path = os.path.join(current_app.config['APPDATA_FOLDER'], file_id)
    if not os.path.exists(file_path):
        return render_template('error.html', msg='File not found')

    # Get an instance of the PadesSignatureExplorer class, used to open/validate
    # PDF signatures.
    sig_explorer = PadesSignatureExplorer()

    # Set PKI default options (see utils.py)
    set_pki_defaults(sig_explorer)

    # Specify that we want to validate the signatures in the file, not only
    # inspect them.
    sig_explorer.validate = True

    # Set the PDF file to be inspected.
    sig_explorer.set_signature_file_from_path(file_path)

    # Call the open() method, which returns the signature file's information.
    signature = sig_explorer.open()

    # Render the signature opening page.
    return render_template('open_pades_express/index.html',
                           signature=signature)
